compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -488,29 +469,9 @@ object RefChecks {
           }*/
     }
 
-    /** We declare a match if either we have a full match including matching names
-     *  or we have a loose match with different target name but the types are the same.
-     *  This leaves two possible sorts of discrepancies to be reported as errors
-     *  in `checkOveride`:
-     *
-     *    - matching names, target names, and signatures but different types
-     *    - matching names and types, but different target names
-     */
-    def considerMatching(sym1: Symbol, sym2: Symbol, self: Type): Boolean =
-      if     sym1.owner.is(JavaDefined, butNot = Trait)
-          && sym2.owner.is(JavaDefined, butNot = Trait)
-      then false // javac already handles these checks
-      else if sym1.isType then true
-      else
-        val sd1 = sym1.asSeenFrom(self)
-        val sd2 = sym2.asSeenFrom(self)
-        sd1.matchesLoosely(sd2)
-          && (sym1.hasTargetName(sym2.targetName)
-             || compatibleTypes(sym1, sd1.info, sym2, sd2.info))
-
     val opc = new OverridingPairs.Cursor(clazz):
```

Comment:
PrepJSInterop also performs some overriding checks using Cursor, so I wonder if it should use a subclass of Cursor with the same logic we're now using here in RefChecks, or perhaps the checking logic in PrepJSInterop should be moved to RefChecks? /cc @sjrd

Reply:
It should probably be the same logic as in RefChecks, yes.

Reply:
OK, I let you experiment with it. But I am going to merge this as is, since I do not think I have the time to look at all code that considers overriding pairs.